### ISSUE TITLE
Provide ability to turn off plugins

### ISSUE THREAD

**Issue by viswaramamoorthy** [apache/skywalking#4578]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ x ] Feature or performance improvement

___
### Question
- What do you want to know?

___
### Bug
- Which version of SkyWalking, OS and JRE?

- Which company or project?

- What happen?
If possible, provide a way for reproducing the error. e.g. demo application, component version.

___
### Requirement or improvement
- Please describe about your requirements or improvement suggestions.
By default Skywalking instruments many plugins that are present in ~/apache-skywalking-apm-bin-es7/agent/plugins directory. In Skywalking v7.0 there are 76 plugin jars available. Not all environments need all of these plugins instrumented. 

Proposal is to provide a property configuration via agent.config which can take a file name that can in turn list the plugins needed for an environment. Below elaborates more on specifics of this request

In agent.config, introduce a new property, say "PluginConfigs", as shown below
….
PluginConfigs=/some-directory/skywalking-plugin.def

and 

Let us say I have /some-directory/skywalking-plugin.def as shown below

rabbitmq-5.x=org.apache.skywalking.apm.plugin.rabbitmq.define.RabbitMQProducerInstrumentation
rabbitmq-5.x=org.apache.skywalking.apm.plugin.rabbitmq.define.RabbitMQConsumerInstrumentation

In this setup, expectation is to just instrument only RabbitMQ consumer and producer although I have agent installed with all 76 plugins.

In summary, new property "PluginConfigs" acts a override to current behavior which instruments all plugins available under "agent/plugin" directory. And "PluginConfigs" property is optional. when "PluginConfigs" is not configured, current behavior to instrument all available plugins based on class path can be retained.

**Comment by wu-sheng** [comment]
Why no just moving the plugin jar files to other places?

**Comment by viswaramamoorthy** [comment]
This feature will help have a common oap-server that can service different applications instrumenting different plugins. For example, lower life cycles (like staging) can have all plugins turned on vs higher life cycles (like production) can have few plugins turned. Also, if problems noticed in higher life cycle, services can be redeployed with updated agent.config/skywalking-plugin.def to enable more plugins as needed. It gives more flexibility.

**Comment by wu-sheng** [comment]
I prefer you still use the agent.config to deactivate, by listing the disable plugin names. You should notice all plugin.def includes names, You should use those names to disable agent plugins. 

WDYT?

And, this discussion should happen in the mail list, could you forward there?

**Comment by viswaramamoorthy** [comment]
Moving discussion to mailing list

**Comment by wu-sheng** [comment]
Update, further discussion happens https://lists.apache.org/thread.html/re13f7a861bbc66f3ee2ab76e709d6fd2cfde44777394346fe98f1959%40%3Cdev.skywalking.apache.org%3E
